darts/models/forecasting/baselines.py

```diff
@@ -4,23 +4,149 @@
 
 A collection of simple benchmark models for univariate series.
 """
-
-from typing import List, Optional, Sequence, Union
+from abc import ABC, abstractmethod
+from typing import List, Optional, Sequence, Tuple, Union
 
 import numpy as np
 
-from darts.logging import get_logger, raise_if, raise_if_not
+from darts.logging import get_logger, raise_if, raise_log
 from darts.models.forecasting.ensemble_model import EnsembleModel
 from darts.models.forecasting.forecasting_model import (
     ForecastingModel,
-    LocalForecastingModel,
+    GlobalForecastingModel,
 )
 from darts.timeseries import TimeSeries
+from darts.utils.utils import seq2series, series2seq
 
 logger = get_logger(__name__)
 
 
-class NaiveMean(LocalForecastingModel):
+class BaselineModel(GlobalForecastingModel, ABC):
+    def __init__(self):
+        super().__init__(add_encoders=None)
+
+    def fit(self, series: Union[TimeSeries, Sequence[TimeSeries]]) -> "BaselineModel":
+        """Fit/train the model on a (or potentially multiple) series.
+        This method is only implemented for naive baseline models to provide a unified fit/predict API with other
+        forecasting models.
+
+        The models are not really trained on the input, but they store the training `series` in case only a single
+        `TimeSeries` was passed. This allows to call `predict()` without having to pass the single `series`.
+
+        All baseline models compute the forecasts for each series directly when calling `predict()`.
+
+        Parameters
+        ----------
+        series
+            One or several target time series. The model will be trained to forecast these time series.
+            The series may or may not be multivariate, but if multiple series are provided
+            they must have the same number of components.
+
+        Returns
+        -------
+        self
+            Fitted model.
+        """
+        series = seq2series(series)
+        super().fit(series=series)
+        self._fit_model(series=series)
+
+    @abstractmethod
+    def _fit_model(self, series: Union[TimeSeries, Sequence[TimeSeries]]) -> None:
+        """Must implement the fit logic and checks for each sub model."""
+        pass
```

Comment:
I think you don't have to write the `pass` (I'm sure for 3.9+ but check for 3.8 :-) )